### PR TITLE
Add egg file to OpenCV 4.6.0 python package for pip + pkgconfig file

### PR DESCRIPTION
**Pull request by Micket**
Add egg-info file to make packages show up in pip

Procedure was stolen from what we have done in VTK

### CODE REVIEW

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.6, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/cef2524c0d7d91fb58c772fb62d63f52 for a full test report.

**Comment by Micket** [comment]
Erh, what am I missing here

This approach is working fine with VTK
```
$ cat /apps/Test2/software/VTK/9.2.2-foss-2022a/lib/python3.10/site-packages/vtk-9.2.2.egg-info 
Metadata-Version: 2.1
Name: vtk
Version: 9.2.2
Summary: VTK is an open-source toolkit for 3D computer graphics, image processing, and visualization
Platform: UNKNOWN

# pip list  | grep vtk
vtk                               9.2.2

# pip show vtk
Name: vtk
Version: 9.2.2
Summary: VTK is an open-source toolkit for 3D computer graphics, image processing, and visualization
Home-page: 
Author: 
Author-email: 
License: 
Location: /apps/Test2/software/VTK/9.2.2-foss-2022a/lib/python3.10/site-packages
Requires: 
Required-by: 
```

and with  this PR on OpenCV
```
 cat /apps/Test2/software/OpenCV/4.6.0-foss-2022a-contrib/lib/python3.10/site-packages/opencv-contrib-python-4.6.0.egg-info 
Metadata-Version: 2.1
Name: opencv-contrib-python
Version: 4.6.0
Summary: Wrapper package for OpenCV python bindings.
Platform: UNKNOWN

# pip list | grep opencv
opencv-contrib-python             4.6.0
```

So far so good!
```
$ pip show opencv-contrib-python
WARNING: Package(s) not found: opencv-contrib-python

$ pip show opencv
Name: opencv-contrib-python
Version: 4.6.0
Summary: Wrapper package for OpenCV python bindings.
Home-page: 
Author: 
Author-email: 
License: 
Location: /apps/Test2/software/OpenCV/4.6.0-foss-2022a-contrib/lib/python3.10/site-packages
Requires: 
Required-by: 
```

wtf, it literally doesn't say just "opencv" anywhere.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.7, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/9e47489b2e55a448cfb231642899c121 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.7, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/a6049a54c3e6f47053a86682045009ca for a full test report.

File: easybuild/easyconfigs/o/OpenCV/OpenCV-4.6.0-foss-2022a-CUDA-11.7.0-contrib.eb [hunk 1]
```diff
@@ -82,6 +86,13 @@ configopts = " ".join([
     "-DCUDA_NVCC_FLAGS=--std=c++14",
 ])
 
+# Install a egg-info file so it is more python friendly
+local_egg_info_src = '%(builddir)s/opencv-contrib-python.egg-info'
+local_egg_info_dest = '%(installdir)s/lib/python%(pyshortver)s/site-packages/opencv-contrib-python-%(version)s.egg-info'
```
Comment (akesandgren) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-4.6.0-foss-2022a-CUDA-11.7.0-contrib.eb:91]: Need to change filename here too

File: easybuild/easyconfigs/o/OpenCV/OpenCV-4.6.0-foss-2022a-contrib.eb [hunk 2]
```diff
@@ -78,6 +82,13 @@ configopts = " ".join([
     "-DBUILD_opencv_python2=OFF",
 ])
 
+# Install a egg-info file so it is more python friendly
+local_egg_info_src = '%(builddir)s/opencv-contrib-python.egg-info'
+local_egg_info_dest = '%(installdir)s/lib/python%(pyshortver)s/site-packages/opencv-contrib-python-%(version)s.egg-info'
```
Comment (akesandgren) [inline easybuild/easyconfigs/o/OpenCV/OpenCV-4.6.0-foss-2022a-contrib.eb:87]: And here

**Review by akesandgren — CHANGES_REQUESTED** [review]

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
vera-c1 - Linux Rocky Linux 8.7, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/Micket/0f39249158c87866f7a527ba0ab6cec1 for a full test report.

**Comment by Micket** [comment]
I can find pc file and pip info now, so i think this is good to go

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @Micket!

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen2

**Comment by Micket** [comment]
@boegelbot please test @ jsc-zen2
EB_ARGS='--optarch=""

**Pull request merged**